uaclient/system.py

```diff
@@ -113,6 +113,15 @@ def get_dpkg_arch() -> str:
     return out.strip()
 
 
+@lru_cache(maxsize=None)
+def get_machine_type() -> str:
+    try:
+        out, _ = subp(["systemd-detect-virt"])
+        return out.strip()
+    except exceptions.ProcessExecutionError:
+        return ""
```

Comment:
@alnvdl-work this is what we're planning on sending to the GetAvailableResources endpoint as `machine_type` - is an empty string a reasonable default value?

Reply:
I think it's fine.
@TeodorPt Please state your opinion here :)

Reply:
BTW, I didn't know we were going to do this change for the available resources calculation as well (I thought it was just the machine activity work).

I have a few concerns about how exactly we will categorize machine types, I think just virtualization type as returned by `systemd-detect-virt` is not enough for the long term. @TeodorPt will schedule a call for us to discuss this issue.

Reply:
Excellent point - this is a separate but very similar issue to the machine activity and container detection work we talked about recently. We should meet and discuss if they can be consolidated or use the same/similar mechanisms/field names and such.